syntax = "proto3";

option java_multiple_files = true;
option java_package = "de.bungenstock.boilerplate-grpc";
option java_outer_classname = "BoilplatedProto";
option objc_class_prefix = "HLW";

package echo;

service Greeter {
    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloReply) {}
  
    rpc SayHelloStreamReply (HelloRequest) returns (stream HelloReply) {}
  
    rpc SayHelloBidiStream (stream HelloRequest) returns (stream HelloReply) {}
  }
  
  // The request message containing the user's name.
  message HelloRequest {
    string name = 1;
  }
  
  // The response message containing the greetings
  message HelloReply {
    string message = 1;
  }